src/autopas/LogicHandler.h

```diff
@@ -791,14 +856,44 @@ void LogicHandler<Particle>::checkMinimalSize() const {
   }
 }
 
+template <typename Particle>
+bool LogicHandler<Particle>::getNeighborListsInvalidDoDynamicRebuild() {
+  return _neighborListInvalidDoDynamicRebuild;
+}
+
 template <typename Particle>
 bool LogicHandler<Particle>::neighborListsAreValid() {
-  if (_stepsSinceLastListRebuild >= _neighborListRebuildFrequency or _autoTuner.willRebuildNeighborLists()) {
+  if (_stepsSinceLastListRebuild >= _neighborListRebuildFrequency or _autoTuner.willRebuildNeighborLists() or
+      getNeighborListsInvalidDoDynamicRebuild()) {
     _neighborListsAreValid.store(false, std::memory_order_relaxed);
   }
   return _neighborListsAreValid.load(std::memory_order_relaxed);
 }
 
+template <typename Particle>
+void LogicHandler<Particle>::checkNeighborListsInvalidDoDynamicRebuild() {
+  const auto skin = getContainer().getVerletSkin();
+  // (skin/2)^2
+  const auto halfSkinSquare = skin * skin * 0.25;
+  // The owned particles in buffer are ignored because they do not rely on the structure of the particle containers,
+  // e.g. neighbour list, and these are iterated over using the region iterator. Movement of particles in buffer doesn't
+  // require a rebuild of neighbor lists.
+  AUTOPAS_OPENMP(parallel reduction(or : _neighborListInvalidDoDynamicRebuild))
+  for (auto iter = this->begin(IteratorBehavior::owned | IteratorBehavior::containerOnly); iter.isValid(); ++iter) {
+    const auto distance = iter->calculateDisplacementSinceRebuild();
+    const double distanceSquare = utils::ArrayMath::dot(distance, distance);
+
+    if (distanceSquare >= halfSkinSquare) {
+      _neighborListInvalidDoDynamicRebuild = true;
+    }
```

Comment:
```suggestion
    _neighborListInvalidDoDynamicRebuild |= distanceSquare >= halfSkinSquare;
```
Since you are reducing anyway we could try to remove the `if`. Avoiding branches is always a good idea to increase throughput.